doc/admin/deployment.md

```diff
@@ -75,6 +75,69 @@ for latest information and examples.
 At this point of the process, the servers: and provider: section of the yaml
 file can be left blank and will be populated in the subsequent sections.
 
+#### Auto generate configuration file
+
+DAOS can attempt to produce a server configuration file that makes optimal use
+of hardware on a given set of hosts through the 'dmg config generate' command:
+
+```bash
+$ dmg config generate --help
+ERROR: dmg: Usage:
+  dmg [OPTIONS] config generate [generate-OPTIONS]
+
+Application Options:
+...
+  -l, --host-list=  comma separated list of addresses <ipv4addr/hostname:port>
+...
+
+[generate command options]
+      -p, --num-pmem=  Minimum number of SCM (pmem) devices required per
+                       storage host in DAOS system
+      -n, --num-nvme=  Minimum number of NVMe devices required per storage host
+                       in DAOS system
+      -c, --net-class=[best-available|ethernet|infiniband] Network class
+                       preferred, defaults to best available (default:
+                       best-available)
+```
+
+The command will output recommended config file if supplied requirements are
+met. Requirements will be derived based on the number of NUMA nodes present on
+the hosts if '--num-pmem' is not specified on the commandline.
+
+- '--num-pmem' specifies the number of persistent memory (pmem) block devices
+that must be present on each host (and must match the number and NUMA affinity
+of matching network interfaces).
+This will equal the number of IO servers that will be configured per host.
+If not set on the commandline, default is the number of NUMA nodes detected on
+the host.
+- '--num-nvme' specifies the minimum number of NVMe SSDs present on each host
+per pmem device (usually matches the number of NUMA nodes on each host).
+For each pmem device selected for the generated config, this number of SSDs
+must be bound to the NUMA node that matches the affinity of the pmem device.
+If not set on the commandline, default is "1".
+- '--net-class' specifies preference for network interface class, options are
+'ethernet', 'infiband' or 'best-available'.
+'best-available' will attempt to choose the most performant sets of interfaces
```

Comment:
Do we want to share that the performance data is what is derived from libfabric, as opposed to some other means that we decided to implement?